V0.44.0 changelog: Pauli-based computation (#8842)
PennyLaneAI/pennylane: doc/releases/changelog-0.44.0.md

  
  New tools dedicated to fault-tolerant quantum computing (FTQC) research based on the Pauli-based
  computation framework are now available! With this release, you can express, compile, and inspect 
  workflows written in terms of Pauli product rotations (PPRs) and Pauli product measurements (PPMs), which are 
  the building blocks of Pauli-based computation framework.

* Writing circuits in terms of `Pauli product measurements <https://pennylane.ai/compilation/pauli-product-measurement>`_
  (PPMs) in PennyLane is now possible with the new :func:`~.pauli_measure` function.
  Using this function in tandem with :class:`~.PauliRot` to represent PPRs unlocks surface-code
  FTQC research spurred from `A Game of Surface Codes <http://arxiv.org/abs/1808.02892>`_.
  [(#8461)](https://github.com/PennyLaneAI/pennylane/pull/8461)
  [(#8631)](https://github.com/PennyLaneAI/pennylane/pull/8631)
  [(#8623)](https://github.com/PennyLaneAI/pennylane/pull/8623)
  [(#8663)](https://github.com/PennyLaneAI/pennylane/pull/8663)
  [(#8692)](https://github.com/PennyLaneAI/pennylane/pull/8692)

  The new :func:`~.pauli_measure` function is currently only for analysis on the ``null.qubit`` device,

>   which allows for resource tracking with :func:`~.specs` and circuit inspection with :func:`~.drawer.draw`.

Reviewer: ```suggestion
  The new :func:`~.pauli_measure` function is currently only supported on the ``null.qubit`` device,
  which allows for circuit inspection with :func:`~.specs` and :func:`~.drawer.draw`.
```

Author: Actually, the inspectability with specs works also with the lightning device, so I think its best to leave:
`only for analysis on the `null.qubit` device

                       ╚════╝
  ```

>   By appliying the :func:`~.specs` function to the circuit above, you can easily determine its resources.

Reviewer: ```suggestion
  You can use the :func:`pennylane.specs` function to easily determine the circuit's resources.
```

    ```
    ```pycon
    >>> circuit(1.1)
    [0.60284353-0.36960984j 0.5076425 +0.4922066j ]
    ```

  * Seven transforms for compiling Clifford+T gates, PPRs, and/or PPMs, including func:`~.transforms.to_ppr`,
  :func:`~.transforms.commute_ppr`, :func:`~.transforms.merge_ppr_ppm`,
  :func:`~.transforms.ppr_to_ppm`, :func:`~.transforms.ppm_compilation`,
  :func:`pennylane.transforms.reduce_t_depth`, and :func:`~.transforms.decompose_arbitrary_ppr`.
  
    ```python
    @qml.qjit(target="mlir")
    @qml.transforms.to_ppr
    @qml.qnode(qml.device("null.qubit", wires=2))
    def circuit():
        qml.H(0)
        qml.CNOT([0, 1])
        qml.T(0)
        return qml.expval(qml.Z(0))
    ```
    ```pycon
    >>> print(qml.specs(circuit, level=2)())
    Total wire allocations: 2
    Total gates: 7
    Circuit depth: Not computed

    Gate types:
      PPR-pi/4: 6
      PPR-pi/8: 1

    Measurements:
      expval(PauliZ): 1
    ```

* Directly decomposing Clifford+T gates and other small gates into PPRs is possible via :func:`~.transforms.decompose`
  and enabling graph-based decompositions with :func:`~.decomposition.enable_graph`. This allows for bypassing the

>   :func:`~.transforms.clifford_t_decomposition`, which can be expensive in some cases.

Reviewer: ```suggestion
* Directly decomposing Clifford+T gates and other small gates into PPRs is possible using the :func:`~.transforms.decompose` transform
  with graph-based decompositions enabled (:func:`~.decomposition.enable_graph`). This allows direct decomposition of certain operators without the need to use approximate methods such as those found in the
  :func:`~.transforms.clifford_t_decomposition` transform, which can sometimes be less efficient.
```
